inc/api.php

```diff
@@ -0,0 +1,841 @@
+			'callback' => __NAMESPACE__ . '\\get_available_segments',
+			'permission_callback' => '__return_true',
+		],
+		'schema' => __NAMESPACE__ . '\\get_segments_schema',
+	] );
+
+	register_rest_route( API_NAMESPACE, 'segments/geo', [
+		[
+			'method' => WP_REST_Server::READABLE,
+			'callback' => __NAMESPACE__ . '\\get_geo_segments',
+			'permission_callback' => '__return_true',
+		],
+		'schema' => __NAMESPACE__ . '\\get_geo_segments_schema',
+	] );
+
+	register_rest_route( API_NAMESPACE, 'segments/interests', [
+		[
+			'method' => WP_REST_Server::READABLE,
+			'callback' => __NAMESPACE__ . '\\get_interests_segments',
+			'permission_callback' => '__return_true',
+		],
+		'schema' => __NAMESPACE__ . '\\get_interests_segments_schema',
+	] );
+
+	register_rest_route( API_NAMESPACE, 'config', [
+		[
+			'method' => WP_REST_Server::READABLE,
+			'callback' => __NAMESPACE__ . '\\get_config',
+			'permission_callback' => '__return_true',
+		],
+		'schema' => __NAMESPACE__ . '\\get_config_schema',
+	] );
+
+	register_rest_route( API_NAMESPACE, 'config/geo/allowed', [
+		[
+			'method' => WP_REST_Server::READABLE,
+			'callback' => '\\Pantheon\\EI\\WP\\Geo\\get_geo_allowed_headers',
+			'permission_callback' => '__return_true',
+		],
+		'schema' => __NAMESPACE__ . '\\get_geo_allowed_config_schema',
+	] );
+
+	register_rest_route( API_NAMESPACE, 'config/interest/cookie-expiration', [
+		[
+			'method' => WP_REST_Server::READABLE,
+			'callback' => '\\Pantheon\\EI\\WP\\Interest\\get_cookie_expiration',
+			'permission_callback' => '__return_true',
+		],
+		'schema' => __NAMESPACE__ . '\\get_interest_cookie_expiration_schema',
+	] );
+
+	register_rest_route( API_NAMESPACE, 'config/interest/post-types', [
+		[
+			'method' => WP_REST_Server::READABLE,
+			'callback' => '\\Pantheon\\EI\\WP\\Interest\\get_interest_allowed_post_types',
+			'permission_callback' => '__return_true',
+		],
+		'schema' => __NAMESPACE__ . '\\get_interest_allowed_post_types_schema',
+	] );
+
+	register_rest_route( API_NAMESPACE, 'config/interest/taxonomies', [
+		[
+			'method' => WP_REST_Server::READABLE,
+			'callback' => '\\Pantheon\\EI\\WP\\Interest\\get_interest_taxonomy',
+			'permission_callback' => '__return_true',
+		],
+		'schema' => __NAMESPACE__ . '\\get_interest_allowed_taxonomies_schema',
+	] );
+
+	register_rest_route( API_NAMESPACE, 'config/interest/threshold', [
+		[
+			'method' => WP_REST_Server::READABLE,
+			'callback' => '\\Pantheon\\EI\\WP\\Interest\\get_interest_threshold',
+			'permission_callback' => '__return_true',
+		],
+		'schema' => __NAMESPACE__ . '\\get_interest_threshold_schema',
+	] );
+
+	register_rest_route( API_NAMESPACE, 'user', [
+		[
+			'method' => WP_REST_Server::READABLE,
+			'callback' => __NAMESPACE__ . '\\get_all_user_data',
+			'permission_callback' => '__return_true',
+		],
+		'schema' => __NAMESPACE__ . '\\get_all_user_data_schema',
+	] );
+
+	// Define the user data endpointds.
+	$user_data = [
+		'conn-speed',
+		'conn-type',
```

Comment:
Yes, intentional. Connection speed and connection type, while they come from Fastly's `geo` config data, don't seem like actual geolocation information. It made more sense in the `ei/user` endpoint to give them their own space rather than nest them under `ei/user/geo`